Sources/InstantSearch/Numeric/RatingControl.swift

```diff
@@ -0,0 +1,207 @@
+//
+//  RatingControl.swift
+//  
+//
+//  Created by Vladislav Fitc on 05/11/2020.
+//
+
+import Foundation
+import UIKit
+
+/// A control for setting a rating value.
+///
+/// The rating is represented as multiple points. The maximumValue defines how many points the control renders.
+/// The images used for setup of one point can be set via **emptyImage**,  **partialImage** and **fullImage** properties.
+///
+/// # Rendering rules
+/// The point is rendered as *empty*, if the difference between the integer part of rating value and the point index  is negative.
+///
+/// - Example: for rating 3.5, its integer part is 3, so the 5-th point (index 4) is rendered as empty
+///
+/// The point is rendered as *full*, if the difference between the integer part of rating value and the point index is greater than 1.
+///
+/// - Example: for rating 3.5, its integer part is 3, so the 3-rd point (index 2) is rendered as full
+///
+/// The point is rendered as *partial*, of the if the difference between the integer part of rating value and the point index is equal to 0 and the fractional part of rating value is greater or equal than 0.5. Otherwise it is rendered as empty.
+///
+/// - Example: for rating 3.5 its integer part is 3, the fractional part is 0.5, so the 4-th point (index 3) is rendered as partial while for rating 3.4, the fractional part is 0.4, so the 4-th point is rendered as empty.
+
+public class RatingControl: UIControl {
+      
+  /// The numeric value of the rating control.
+  ///
+  /// The default value for this property is 0.
+  public var value: Double = 0 {
+    didSet {
+      refresh()
+      sendActions(for: .valueChanged)
+    }
+  }
+  
+  /// The highest possible numeric value for the rating control.
+  ///
+  /// Must be numerically less than maximumValue. If you attempt to set a value equal to or greater than maximumValue, the system raises an invalidArgumentException exception.
```

Comment:
potentially a UT that can be written

Reply:
Actually, that's an erroneous comment copy-pasted from Apple UIStepper documentation :) To remove.